library/core/src/mem/maybe_uninit.rs

```diff
@@ -632,14 +683,14 @@ impl<T> MaybeUninit<T> {
     ///
     /// On top of that, all additional invariants of the type `T` must be
     /// satisfied, as the `Drop` implementation of `T` (or its members) may
-    /// rely on this. For example, a `1`-initialized [`Vec<T>`] is considered
-    /// initialized (under the current implementation; this does not constitute
-    /// a stable guarantee) because the only requirement the compiler knows
-    /// about it is that the data pointer must be non-null. Dropping such a
-    /// `Vec<T>` however will cause undefined behaviour.
+    /// rely on this. For example, setting [`Box<T>`] to an invalid but non zero
+    /// address makes the pointer initialized (under the current implementation;
+    /// this does not constitute a stable guarantee), because the only
+    /// requirement the compiler knows about it is that the data pointer must be
+    /// non-null. Dropping such a `Box<T>` however will cause undefined behaviour.
```

Comment:
That's not true, `Box` is `dereferencable` so it is UB to create a dangling `Box` even if you never use it.

Reply:
I've changed it back to vec.